common/src/main/java/org/wso2/testgrid/common/DashboardSetup.java

```diff
@@ -99,50 +108,70 @@ private String getDataSource(String name) {
     }
 
     /**
-     * This method will create the header of the POST request
-     * @param restUrl url of the grafana server
-     * @param apikey APIkey of the grafana server
-     * @return return a http post request
+     * This method will create a grafana datasource according to TestPlanID
      */
-    private HttpPost createConnectivity(String restUrl, String apikey) {
-        HttpPost post = new HttpPost(restUrl + ":3000/api/datasources/");
-        post.setHeader("AUTHORIZATION", apikey);
-        post.setHeader("Content-Type", ContentType.APPLICATION_JSON.toString());
-        post.setHeader("Accept", ContentType.APPLICATION_JSON.toString());
-        return post;
-    }
+    public void addGrafanaDataSource() {
 
-    /**
-     * This command will handle the posting http request to REST  api
-     * @param jsonData json string of the data source
-     * @param httpPost post request with authentication
-     */
-    private void executeReq(String jsonData, HttpPost httpPost) {
+        DataOutputStream dataOutputStream = null;
         try {
-            executeHttpRequest(jsonData, httpPost);
-        } catch (UnsupportedEncodingException e) {
-            logger.error(StringUtil.concatStrings("error while encoding grafana http api url : ", e));
-        } catch (IOException e) {
-            logger.error(StringUtil.concatStrings("ioException occured while sending http request : ", e));
+            String url = "https://" + ConfigurationContext.getProperty(ConfigurationContext.ConfigurationProperties
+                    .GRAFANA_DATASOURCE) + ":3000/api/datasources/";
+
+            HostnameVerifier allHostsValid = (hostname, session) -> true;
+            HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
+
+            URL obj = new URL(url);
+            HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
+            //add request header
+            con.setRequestMethod("POST");
+            con.setRequestProperty("User-Agent", USER_AGENT);
+            con.setRequestProperty("Content-Type", ContentType.APPLICATION_JSON.toString());
+            con.setRequestProperty("Accept", ContentType.APPLICATION_JSON.toString());
+            con.setRequestProperty("Authorization", apikey);
+            SSLSocketFactory sslSocketFactory = createSslSocketFactory();
+            con.setSSLSocketFactory(sslSocketFactory);
+            String urlParameters = getDataSource(testplanID);
+            con.setDoOutput(true);
+            dataOutputStream = new DataOutputStream(con.getOutputStream());
+            dataOutputStream.writeBytes(urlParameters);
+            dataOutputStream.flush();
+            int responseCode = con.getResponseCode();
+            if (responseCode == 200) {
```

Comment:
Is it possible to refer the 200 value from a relevant reference class?
Ex:  `javax.ws.rs.core.Response.Status.Ok`

Reply:
will need to add new dependancies